File: src/main/java/com/domclick/config/FlywayConfig.java
package com.domclick.config;

import org.flywaydb.core.Flyway;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import javax.sql.DataSource;

@Configuration
public class FlywayConfig {
    @Autowired
    DataSource dataSource;

    @Bean
    public Flyway flyway(){
        Flyway flyway = new Flyway();
        flyway.setDataSource(dataSource);
        flyway.setSqlMigrationPrefix("V");
        flyway.migrate();
        return  flyway;

    }
}

File: app/src/main/kotlin/com/domclick/config/FlywayConfig.kt
package com.domclick.config

import com.domclick.config.properties.persistance.PersistenceProperties
import org.flywaydb.core.Flyway
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import javax.sql.DataSource

@Configuration
class FlywayConfig(
        private val dataSource_: DataSource,
        private val persistenceProperties: PersistenceProperties
) {

    @Bean
    fun flyway() = Flyway().apply {
        this.dataSource = dataSource_
        sqlMigrationPrefix = "V"
        setLocations("db.migration.${persistenceProperties.vendor}")
        migrate()
    }
}
